### PR TITLE
S03 us01 navbar

### PR DESCRIPTION
**Pull request by ozen2**

### CODE REVIEW

**Review by axeljdev — APPROVED** [review]

**Review by JulienChevreauPro — APPROVED** [review]

**Review by adrale — APPROVED** [review]

**Review by adrale — APPROVED** [review]

**Review by JulienChevreauPro — APPROVED** [review]

**Review by adrale — APPROVED** [review]

File: client/src/components/Navbar.jsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import { useState, useRef } from "react";
+import { Link } from "react-router-dom";
+import NavAccess from "../services/NavAccess";
+
+function Navbar() {
+  const [isOpen, setIsOpen] = useState(false);
+  const [isVisible, setIsVisible] = useState(false);
+  const menuRef = useRef(null);
+
+  const handleChange = () => {
```
Comment (Yoann-Fortin) [inline client/src/components/Navbar.jsx:10]: Extract this logic in parent's component will be better

File: client/src/components/Navbar.jsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        className="block absolute left-4 top-4 md:hidden"
+        onClick={handleChange}
+        aria-label="Ouvrir le menu"
+      >
+        <img src="/src/assets/images/menuBurger.svg" alt="icone menu burger" />
+      </button>
+      <nav
+        ref={menuRef}
+        className={`
+          ${
```
Comment (Yoann-Fortin) [inline client/src/components/Navbar.jsx:37]: Same here

File: client/src/services/NavAccess.jsx
Comment (Yoann-Fortin) [inline client/src/services/NavAccess.jsx:1]: Isn't a component, a class will be a good option here

**Review by Yoann-Fortin — CHANGES_REQUESTED** [review]

File: client/src/services/NavAccess.jsx
Comment (Yoann-Fortin) [inline client/src/services/NavAccess.jsx:1]: Isn't a JSX file

File: client/src/services/NavAccess.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      } if (!e.shiftKey && document.activeElement === lastElement) {
+        e.preventDefault();
+        firstElement.focus();
+      }
+    }
+  };
+
+  addEventListener() {
+    const menuRefCurrent = this.menuRef.current;
+    if (!menuRefCurrent) return;
```
Comment (Yoann-Fortin) [inline client/src/services/NavAccess.jsx:28]: A condition with a true value would be better

File: client/src/services/NavAccess.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    menuRefCurrent.addEventListener("keydown", this.handleTabKey);
+  }
+
+  removeEventListener() {
+    const menuRefCurrent = this.menuRef.current;
+    if (!menuRefCurrent) return;
```
Comment (Yoann-Fortin) [inline client/src/services/NavAccess.jsx:34]: Same

File: client/src/services/NavAccess.jsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+class NavAccess {
+  constructor(menuRef) {
+    this.menuRef = menuRef;
+    this.focusableElementsString = "a, button";
+  }
+
+  handleTabKey = (e) => {
+    const menuRefCurrent = this.menuRef.current;
+    if (!menuRefCurrent) return;
```
Comment (Yoann-Fortin) [inline client/src/services/NavAccess.jsx:9]: A condition with a true value would be better

**Pull request merged**